pkg/fleetautoscalers/fleetautoscalers.go

```diff
@@ -41,18 +41,92 @@ var client = http.Client{
 
 // computeDesiredFleetSize computes the new desired size of the given fleet
 func computeDesiredFleetSize(fas *autoscalingv1.FleetAutoscaler, f *agonesv1.Fleet) (int32, bool, error) {
-
 	switch fas.Spec.Policy.Type {
 	case autoscalingv1.BufferPolicyType:
 		return applyBufferPolicy(fas.Spec.Policy.Buffer, f)
 	case autoscalingv1.WebhookPolicyType:
 		return applyWebhookPolicy(fas.Spec.Policy.Webhook, f)
 	}
 
-	return f.Status.Replicas, false, errors.New("wrong policy type, should be one of: Buffer, Webhook")
+	return 0, false, errors.New("wrong policy type, should be one of: Buffer, Webhook")
 }
 
-func applyWebhookPolicy(w *autoscalingv1.WebhookPolicy, f *agonesv1.Fleet) (int32, bool, error) {
+func buildURLFromWebhookPolicy(w *autoscalingv1.WebhookPolicy) (*url.URL, error) {
+	if w.URL != nil {
+		if *w.URL == "" {
+			return nil, errors.New("URL was not provided")
+		}
+
+		return url.ParseRequestURI(*w.URL)
+	}
+
+	if w.Service.Name == "" {
+		return nil, errors.New("service name was not provided")
+	}
+
+	if w.Service.Path == nil {
+		empty := ""
+		w.Service.Path = &empty
+	}
+
+	if w.Service.Namespace == "" {
+		w.Service.Namespace = "default"
+	}
+
+	scheme := "http"
+	if w.CABundle != nil {
+		scheme = "https"
+
+		if err := setCABundle(w.CABundle); err != nil {
+			return nil, err
+		}
+	}
+
+	return createURL(scheme, w.Service.Name, w.Service.Namespace, *w.Service.Path), nil
+}
+
+// moved to a separate method to cover it with unit tests and check that URL corresponds to a proper pattern
+func createURL(scheme, name, namespace, path string) *url.URL {
+	return &url.URL{
+		Scheme: scheme,
+		Host:   fmt.Sprintf("%s.%s.svc:8000", name, namespace),
+		Path:   path,
+	}
+}
+
+func setCABundle(caBundle []byte) error {
+	// We can have multiple fleetautoscalers with different CABundles defined,
+	// so we switch client.Transport before each POST request
+	rootCAs := x509.NewCertPool()
+	if ok := rootCAs.AppendCertsFromPEM(caBundle); !ok {
+		return errors.New("no certs were appended from caBundle")
+	}
+	client.Transport = &http.Transport{
+		TLSClientConfig: &tls.Config{
+			RootCAs: rootCAs,
+		},
+	}
+	return nil
+}
+
+func applyWebhookPolicy(w *autoscalingv1.WebhookPolicy, f *agonesv1.Fleet) (replicas int32, limited bool, err error) {
+	if w == nil {
+		return 0, false, errors.New("nil WebhookPolicy passed")
+	}
+
+	if f == nil {
+		return 0, false, errors.New("nil Fleet passed")
```

Comment:
I would prefer error message in this form:
```suggestion
		return 0, false, errors.New("fleet parameter must not be nil")
```